Examples/E2/Notification.cs

```diff
@@ -20,19 +25,30 @@ public void PostMessageToChannel(string channel, string message)
     {
         Console.WriteLine($"Message sent to Slack channel {channel}: {message}");
     }
-} 
+}
 #endregion
 
+public class SlackNotidicationService(string channel) : INotificationService
+{
+    private readonly SlackNotification _slackNotification;
+    public void Send(string message)
+    {
+        _slackNotification.PostMessageToChannel(channel, message);
+    }
+}
+
+
+
 public class NotificationService()
 {
-    public ... Get(string type)
+    public INotificationService Get(string type, string channel = null)
```

Comment:
if we have another 3rd party which wants key string parameter, will you add another parameter?